lib/scala/editions/src/main/scala/org/enso/editions/EditionResolver.scala

```diff
@@ -26,7 +27,15 @@ case class EditionResolver(provider: EditionProvider) {
   def resolve(
     edition: RawEdition
   ): Either[EditionResolutionError, ResolvedEdition] =
-    resolveEdition(edition, Nil)
+    resolveEdition(edition, Nil).left.flatMap { err =>
+      LoggerFactory
+        .getLogger(classOf[EditionResolver])
+        .warn(
+          "Failed to resolve original edition. Trying fallback to the default one",
+          err
+        )
+      resolveEdition(DefaultEdition.getDefaultEdition, Nil).left.map(_ => err)
```

Comment:
I think the `resolve` method logic should fail returning `Left` as designed, and the fallback should be added when resolving the project edition. Otherwise, `Either` in the method signature has little sense.
Or there should be another `resolve` method returning `ResolvedEdition`, specifying that it will fallback to the default one

Reply:
OK, sounds reasonable. I will make the change